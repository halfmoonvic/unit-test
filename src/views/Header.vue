<template>
  <div>
    <input
      type="text"
      data-test="input"
      v-model="inputValue"
      @keyup.enter="addTodoItem"
    />
  </div>
</template>

<script lang="ts">
// third party tools and libraries
import Vue from 'vue';
import { Component } from 'vue-property-decorator';
// local functions and variables
// local components

@Component
export default class Header extends Vue {
  private inputValue: string = '';
  private addTodoItem() {
    if (!this.inputValue) {
      return;
    }
    this.$emit('add', this.inputValue);
    this.inputValue = '';
  }
}
</script>

<style lang="scss" scoped></style>
