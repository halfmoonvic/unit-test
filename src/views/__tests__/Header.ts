import { shallowMount } from '@vue/test-utils';
import Header from '@/views/Header.vue';

describe('Header.vue', () => {
  it('包含 input 框', () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    expect(input.exists()).toBeTruthy();
  });

  it('input 框初始值为空', () => {
    const wrapper = shallowMount(Header);
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe('');
  });

  it('input 框值发生变化，数据应该跟着变', () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    input.setValue('dell lee');
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe('dell lee');
  });

  it('input 框 输入回车，无内容时，无反应', () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    input.setValue('');
    input.trigger('keyup.enter');
    expect(wrapper.emitted('add')).toBeFalsy();
  });

  it('input 框 输入回车，有内容时，向外触发事件,同时清空 inputValue', () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    input.setValue('todo one');
    input.trigger('keyup.enter');
    expect(wrapper.emitted('add')).toBeTruthy();
    expect(wrapper.vm.$data.inputValue).toBe('');
  });

  it('样式发生改变，做提示', () => {
    const wrapper = shallowMount(Header);
    expect(wrapper).toMatchSnapshot();
  });
});
