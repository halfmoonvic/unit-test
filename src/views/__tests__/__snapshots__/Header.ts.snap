// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue 样式发生改变，做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input type="text" data-test="input" placeholder="TotoItem" class="header-input"></div>
</div>
`;
